lighthouse-core/computed/image-records.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('./computed-artifact.js');
+
+class ImageRecords {
+  /**
+   * @param {LH.Artifacts.NetworkRequest[]} networkRecords
+   */
+  static indexNetworkRecords(networkRecords) {
+    return networkRecords.reduce((map, record) => {
+      // An image response in newer formats is sometimes incorrectly marked as "application/octet-stream",
+      // so respect the extension too.
+      const isImage = /^image/.test(record.mimeType) || /\.(avif|webp)$/i.test(record.url);
+      // The network record is only valid for size information if it finished with a successful status
+      // code that indicates a complete image response.
+      if (isImage && record.finished && record.statusCode === 200) {
+        map[record.url] = record;
+      }
+
+      return map;
+    }, /** @type {Record<string, LH.Artifacts.NetworkRequest>} */ ({}));
+  }
+
+  /**
+   * @param {{ImageElements: LH.Artifacts['ImageElements'], networkRecords: LH.Artifacts.NetworkRequest[]}} data
```

Comment:
we *typically* have computed artifacts accept the raw devtools log and let them re-request the network records. 

don't know if that makes it any easier or if you have a strong opinion about that pattern, and we don't follow it consistently (processedTrace is a good recent example), but it makes some sense to me :)

Reply:
`ByteEfficieny` audits pass `networkRecords` directly to the `audit_` function. See `uses-responsive-images.js`.

Reply:
I was expecting more audits to use this than seen here, but I'm OK with this.